07-contributor-guide.Rmd

```diff
@@ -24,11 +33,10 @@ Following good software development and coding practices simplifies collaboratio
 * Use consistent formatting and indentation to improve readability and organization
 * Group code into separate blocks for individual tasks
 * Avoid hard-coded values to ensure portability of code
-* Follow the DRY principle - "Don't Repeat Yourself" (or your code)
+* Follow the "Don't Repeat Yourself" (DRY) principle for coding
 * Avoid deep nesting
-* Limit line length (wrap ~72 characters)
+* Limit line length (wrap ~72 characters) of code but use a single line per paragraph in this markdown document
 * Capitalize SQL queries so they are readily distinguishable from table/column names
-* Lint your code
 
 ## Roadmap to FIMS File Structure and Organization
```

Comment:
Do we want to document the FIMS file structure here in the collaborative workflow? It might be easier to remember to maintain the descriptions if we keep them within the FIMS repo. We can create short README.md files in the distributions, interface, and other folders to provide brief introductions (see [googletest example](https://github.com/google/googletest/tree/main/googlemock)). However, I'm not sure if these README.md files follow good practices for developing R packages.

Reply:
I am open to ideas here. I talked to @k-doering-NOAA about potentially creating a GitHub action that creates an issue in the collaborative_workflow repo when folders are added to the FIMS file structure. But that might be overly complicated. @Bai-Li-NOAA what do you think would be the best for training purposes? If we had documentation in the folders themselves that we could "scrape" when making presentations or training material I think that would be ideal. I know that we can add files to .Rbuildignore so they are ignored when building FIMS if we want READMEs in the directories themselves.

Reply:
I like the idea of automatically updating training materials to introduce the FIMS file structure. We could include short README.md files in each folder and use a vignette to automatically compile all *.md files from inst/include. Additionally, we might consider using Doxygen to generate a [file dependencies diagram](https://www.doxygen.nl/manual/diagrams.html) to be included in the vignette.

Reply:
So how should we proceed? Should we leave the text for this section as is and turn your comment above into a new issue? Or, should we delete the text and turn your comment above into a new issue? Or, maybe some other option.

Reply:
Maybe keep the text for this section as is for now. I can create a new issue to explore moving the text into Markdown files and automatically generating a vignette from them. I can help with setting up the vignette and testing its functionality.

Reply:
Alright, sounds good.